commit 74aaf96feaca80285912cc6f19575b3e97177918
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Tue Feb 22 13:10:52 2022 -0500

    SUNRPC: Teach server to recognize RPC_AUTH_TLS
    
    Initial support for the RPC_AUTH_TLS authentication flavor enables
    NFSD to eventually accept an RPC_AUTH_TLS probe from clients. This
    patch simply prevents NFSD from rejecting these probes completely.
    
    In the meantime, graft this support in now so that RPC_AUTH_TLS
    support keeps up with generic code and API changes in the RPC
    server.
    
    Down the road, server-side transport implementations will populate
    xpo_start_tls when they can support RPC-with-TLS. For example, TCP
    will eventually populate it, but RDMA won't.
    
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>

diff --git a/include/linux/sunrpc/svc_xprt.h b/include/linux/sunrpc/svc_xprt.h
index 42e113742429..20068ccfd0cc 100644
--- a/include/linux/sunrpc/svc_xprt.h
+++ b/include/linux/sunrpc/svc_xprt.h
@@ -28,6 +28,7 @@ struct svc_xprt_ops {
 	void		(*xpo_free)(struct svc_xprt *);
 	void		(*xpo_secure_port)(struct svc_rqst *rqstp);
 	void		(*xpo_kill_temp_xprt)(struct svc_xprt *);
+	void		(*xpo_start_tls)(struct svc_xprt *);
 };
 
 struct svc_xprt_class {